tests/Unit/Caching/SimpleStringCacheTest.php

```diff
@@ -0,0 +1,153 @@
+        $this->expectExceptionMessage('Please provide a non-empty key.');
+
+        $this->subject->has('');
+    }
+
+    /**
+     * @test
+     */
+    public function hasInitiallyIsFalseForAnyKey(): void
+    {
+        self::assertFalse($this->subject->has('hello'));
+    }
+
+    /**
+     * @test
+     */
+    public function hasForKeyThatHasNotBeenSetReturnsFalse(): void
+    {
+        $this->subject->set('hello', 'world');
+
+        self::assertFalse($this->subject->has('what'));
+    }
+
+    /**
+     * @test
+     */
+    public function hasForKeyThatHasBeenSetReturnsTrue(): void
+    {
+        $key = 'hello';
+        $this->subject->set($key, 'world');
+
+        self::assertTrue($this->subject->has($key));
+    }
+
+    /**
+     * @test
+     */
+    public function hasForKeyThatHasBeenSetAndOverwrittenReturnsTrue(): void
+    {
+        $key = 'hello';
+        $this->subject->set($key, 'world');
+        $this->subject->set($key, 'PHP');
+
+        self::assertTrue($this->subject->has($key));
+    }
+
+    /**
+     * @test
+     */
+    public function getForEmptyKeyThrowsException(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionCode(1625995840);
+        $this->expectExceptionMessage('Please provide a non-empty key.');
+
+        $this->subject->get('');
+    }
+
+    /**
+     * @test
+     */
+    public function setForEmptyKeyThrowsException(): void
+    {
+        $this->expectException(\InvalidArgumentException::class);
+        $this->expectExceptionCode(1625995840);
+        $this->expectExceptionMessage('Please provide a non-empty key.');
+
+        $this->subject->set('', 'some value');
+    }
+
+    /**
+     * @test
+     */
+    public function getForKeyThatHasNotBeenSetThrowsException(): void
+    {
+        $this->expectException(\BadMethodCallException::class);
+        $this->expectExceptionMessage('You can only call `get` with a key for an existing value.');
+        $this->expectExceptionCode(1625996246);
+
+        $this->subject->get('hello');
+    }
+
+    /**
+     * @test
+     */
+    public function getForKeyThatHasBeenSetReturnsSetValue(): void
+    {
+        $key = 'hello';
+        $value = 'world';
+
+        $this->subject->set($key, $value);
+
+        self::assertSame($value, $this->subject->get($key));
+    }
+
+    /**
+     * @test
+     */
+    public function getForKeyThatHasBeenSetTwiceReturnsValueThatHasBeenSetLast(): void
+    {
+        $key = 'hello';
+        $value = 'world';
+
+        $this->subject->set($key, 'coffee');
+        $this->subject->set($key, $value);
+
+        self::assertSame($value, $this->subject->get($key));
+    }
+
+    /**
+     * @test
+     */
+    public function setAndGetCanAlsoReturnEmptyString(): void
+    {
+        $key = 'hello';
+        $value = '';
+
+        $this->subject->set($key, $value);
+
+        self::assertSame($value, $this->subject->get($key));
+    }
```

Comment:
I think this should be combined with `getForKeyThatHasBeenSetReturnsSetValue`, with a data provider providing a non-empty string and an empty string.